### ISSUE TITLE
[Suggestions] Enhancement and Features Suggestions

### ISSUE THREAD

**Issue by iyrepl** [ATechnoHazard/katbin#14]
1.Add Redis to raise its perfomance.
2.Add Postgre SQL support.
3.Add backend web admin panel to make the settings like deletion, data volume, login permission management
4.Add the shorten URL expired time function.
6.Add Statistics Function:hits statistics, ip regions and devices used by users.
7.Add password Function: allow users to enter password to continue access.
A little troublesome, but I wish it would help more people(^▽^ )

**Comment by evanferrao** [comment]
> 6.Add Statistics Function:hits statistics, ip regions and devices used by users. 

I _think_ this feature was _once present_. I'm not sure though. I have faint memories of using a pastebin with statistics that showed the number of people who viewed the paste along with their IP regions. I'm not sure if it was katbin or not, but the UI was very similar. If it was katbin, I don't know why it was removed.

**Comment by SphericalKat** [comment]
> I think this feature was once present. I'm not sure though. I have faint memories of using a pastebin with statistics that showed the number of people who viewed the paste along with their IP regions. I'm not sure if it was katbin or not, but the UI was very similar. If it was katbin, I don't know why it was removed.

You are correct, it was Katbin. I received a bunch of feedback from users who didn't want analytics, so it was removed.

As for the feature requests, I will begin work on some of them.

On a side note, I welcome pull requests implementing features :)
